src/axom/quest/SamplingShaper.hpp

```diff
@@ -378,44 +145,77 @@ class SamplingShaper : public Shaper
       return;
     }
 
-    SLIC_INFO(axom::fmt::format("{:-^80}", " Generating the octree "));
+    SLIC_INFO(axom::fmt::format("{:-^80}", " Generating the spatial index "));
 
     const auto& shapeName = shape.getName();
 
-    switch(shapeDimension)
+    // note: ignoring the global shapeDimension for now since it's causing problems
+    // reading c2c when the dimension is Three
+    AXOM_UNUSED_VAR(shapeDimension);
+    if(this->shapeFormat(shape) == "c2c")
```

Comment:
Note: This class is screaming for `std::variant` -- we have a bunch of implementations w/ a common API, but only one of them can be valid for the instance.  Really looking forward to when we can use `C++17` features...

Reply:
Good ol' polymorphism still works too and would allow a single shaper implementation pointer, which sounds like what you want from this factory.